nntrainer/layers/layer_context.cpp

```diff
@@ -308,4 +333,86 @@ bool RunLayerContext::readyToUse() const {
   return !inputs[0]->getVariable().empty();
 }
 
+/**
+ * @brief   validates the run context after run
+ *
+ * @return true if ready, else false
+ */
+bool RunLayerContext::validate(bool skip_input, bool skip_label) {
+  /**
+   * @note a common mistake when using run_context is re-assigning the tensor
+   * references which leads to nasty bugs. This validation ensures that the
+   * tensors are not set mistakenly by verifying their unique names
+   */
+#ifdef ENABLE_TEST
```

Comment:
How about using special option to check this? we commonly use ENABLE_TEST in packagin, I think it is easy to mistakenly use this in optimized build.